src/wallet/message_store.cpp

```diff
@@ -392,6 +403,43 @@ void message_store::process_auto_config_data_message(uint32_t id)
   signer.auto_config_running = false;
 }
 
+void add_hash(crypto::hash &sum, const crypto::hash &summand)
+{
+  for (uint32_t i = 0; i < crypto::HASH_SIZE; ++i)
+  {
+    uint32_t x = (uint32_t)sum.data[i];
+    uint32_t y = (uint32_t)summand.data[i];
+    sum.data[i] = (char)((x + y) % 256);
+  }
+}
+
+// Calculate a checksum that allows signers to make sure they work with an identical signer config
+// by exchanging and comparing checksums out-of-band i.e. not using the MMS;
+// Because different signers have a different order of signers in the config work with "adding"
+// individual hashes because that operation is commutative
+std::string message_store::get_config_checksum() const
+{
+  crypto::hash sum = crypto::null_hash;
+  add_hash(sum, crypto::cn_fast_hash(&m_num_authorized_signers, sizeof(m_num_authorized_signers)));
+  add_hash(sum, crypto::cn_fast_hash(&m_num_required_signers, sizeof(m_num_required_signers)));
+  for (uint32_t i = 0; i < m_num_authorized_signers; ++i)
+  {
+    const authorized_signer &m = m_signers[i];
+    add_hash(sum, crypto::cn_fast_hash(m.transport_address.data(), m.transport_address.size()));
+    if (m.monero_address_known)
+    {
+      add_hash(sum, crypto::cn_fast_hash(&m.monero_address.m_spend_public_key, sizeof(m.monero_address.m_spend_public_key)));
+      add_hash(sum, crypto::cn_fast_hash(&m.monero_address.m_view_public_key, sizeof(m.monero_address.m_view_public_key)));
+    }
+  }
```

Comment:
Yes, that's the idea, only summing hashes. Your "sort first, then hash" alternative idea is good and would work, but I think it's safe enough for the purpose as it is now.

Reply:
It means ordering is not "hashed in". Not sure if that can be a problem or not. If not, then I think it's good.

Reply:
Ah, I see. No, the ordering is not among the things that must get enforced / checked for "same". Everybody in fact will have a different order because the signer "me" is fixed to 0 and of course different from wallet to wallet.